### PR TITLE
Update for kubernetes 1.20 support

### PR DESCRIPTION
**Pull request by evrardjp**
This ensures we bump the code for 1.20.
It updates the testing to ensure kured works on a 1.20 cluster,
removes the testing on 1.17 (as it is now deprecated).
Libraries remain on 1.19, to avoid breaking 1.18 clusters.

Related: #267 .

### CODE REVIEW

File: charts/kured/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: "1.5.1"
 description: A Helm chart for kured
 name: kured
-version: 2.2.2
+version: 2.2.3
```
Comment (dholbach) [inline charts/kured/Chart.yaml:5]: 2.2.2 is not released yet, so this is not necessary AFAICS: https://github.com/weaveworks/kured/tree/gh-pages

File: charts/kured/templates/clusterrole.yaml [hunk 2]
```diff
@@ -9,9 +9,9 @@ rules:
 # Allow kured to read spec.unschedulable
 # Allow kubectl to drain/uncordon
 #
-# NB: These permissions are tightly coupled to the bundled version of kubectl; the ones below
-# match https://github.com/kubernetes/kubernetes/blob/v1.19.4/staging/src/k8s.io/kubectl/pkg/cmd/drain/drain.go
```
Comment (dholbach) [inline charts/kured/templates/clusterrole.yaml:13]: Is the link useful for figuring out which apigroups, etc? If yes, we should probably keep it around?

Comment (evrardjp) [reply charts/kured/templates/clusterrole.yaml:13]: Isn't the other text good enough? I am reluctant to keep it to a single version, but I think I can live with it.

Comment (evrardjp) [reply charts/kured/templates/clusterrole.yaml:13]: Ok let me rollback that change, and keep it as is.

File: charts/kured/Chart.yaml (hunk 1 above)
Comment (evrardjp) [reply charts/kured/Chart.yaml:5]: does it matter? the code with a helm chart of 2.2.2 was effectively out in the wild, even though not published through our gh pages...

Which reminds me that we should probably publish the helm chart on merge in master?

File: charts/kured/templates/clusterrole.yaml (hunk 2 above)
Comment (evrardjp) [reply charts/kured/templates/clusterrole.yaml]: Done.

File: charts/kured/Chart.yaml (hunk 1 above)
Comment (dholbach) [reply charts/kured/Chart.yaml:5]: No, it needs a tag to publish.

cc @ckotzbauer

Comment (evrardjp) [reply charts/kured/Chart.yaml:5]: I have a PR to fix this: https://github.com/weaveworks/kured/pull/269

Comment (ckotzbauer) [reply charts/kured/Chart.yaml:5]: I reviewed #269. That's a good change for faster chart releases :+1:

**Review by dholbach — APPROVED** [review]

**Pull request merged**